app/helpers/application_helper/toolbar_builder.rb

```diff
@@ -1272,9 +1272,9 @@ def build_toolbar_disable_button(id)
       end
     when "Zone"
       case id
-      when "collect_logs", "collect_current_logs"
-        return "Cannot collect current logs unless there are started #{ui_lookup(:tables => "miq_servers")} in the Zone" if @record.miq_servers.collect { |s| s.status == "started" ? true : nil }.compact.length == 0
-        return "This Zone and one or more active #{ui_lookup(:tables => "miq_servers")} in this Zone do not have Log Depot settings configured, collection not allowed" if @record.miq_servers.select(&:log_depot).blank?
+      when "zone_collect_logs", "zone_collect_current_logs"
+        return "Cannot collect current logs unless there are started #{ui_lookup(:tables => "miq_servers")} in the Zone" if @record.none_miq_server_started?
```

Comment:
:+1:  Thank you for extracting this logic out of the controller and into a model method.